### ISSUE TITLE
Paper 1.21.1 build 37 crashes on servers with configuration from older paper versions

### ISSUE THREAD

**Issue by JulianVennen** [PaperSpigot/Paper#11298]
### Stack trace

```
[12:30:32] [Server thread/ERROR]: Encountered an unexpected exception
org.spongepowered.configurate.serialize.SerializationException: [item-validation, book-size, page-max] of type io.papermc.paper.configuration.type.number.IntOr$Disabled: No TypeSerializer found for field pageMax of type class io.papermc.paper.configuration.type.number.IntOr$Disabled
	at org.spongepowered.configurate.objectmapping.FieldData.serializerFrom(FieldData.java:138) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at org.spongepowered.configurate.objectmapping.ObjectMapperImpl.load0(ObjectMapperImpl.java:71) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at org.spongepowered.configurate.objectmapping.ObjectMapperImpl.load(ObjectMapperImpl.java:56) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at org.spongepowered.configurate.objectmapping.ObjectMapperFactoryImpl.deserialize(ObjectMapperFactoryImpl.java:230) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at org.spongepowered.configurate.serialize.TypeSerializer.deserialize(TypeSerializer.java:96) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at org.spongepowered.configurate.objectmapping.ObjectMapperImpl.load0(ObjectMapperImpl.java:76) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at org.spongepowered.configurate.objectmapping.ObjectMapperImpl.load(ObjectMapperImpl.java:56) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at org.spongepowered.configurate.objectmapping.ObjectMapperFactoryImpl.deserialize(ObjectMapperFactoryImpl.java:230) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at org.spongepowered.configurate.serialize.TypeSerializer.deserialize(TypeSerializer.java:96) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at org.spongepowered.configurate.objectmapping.ObjectMapperImpl.load0(ObjectMapperImpl.java:76) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at org.spongepowered.configurate.objectmapping.ObjectMapperImpl.load(ObjectMapperImpl.java:56) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at org.spongepowered.configurate.objectmapping.ObjectMapperFactoryImpl.deserialize(ObjectMapperFactoryImpl.java:230) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at org.spongepowered.configurate.AbstractConfigurationNode.get0(AbstractConfigurationNode.java:203) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at org.spongepowered.configurate.AbstractConfigurationNode.get(AbstractConfigurationNode.java:[16](https://mclo.gs/6rp0D7Z#L16)1) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at org.spongepowered.configurate.ConfigurationNode.get(ConfigurationNode.java:521) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at org.spongepowered.configurate.ConfigurationNode.require(ConfigurationNode.java:408) ~[configurate-core-4.2.0-SNAPSHOT.jar:?]
	at io.papermc.paper.configuration.Configurations.lambda$creator$0(Configurations.java:91) ~[paper-1.21.1.jar:1.21.1-37-ba1b016]
	at io.papermc.paper.configuration.Configurations.initializeGlobalConfiguration(Configurations.java:138) ~[paper-1.21.1.jar:1.21.1-37-ba1b016]
	at io.papermc.paper.configuration.Configurations.initializeGlobalConfiguration(Configurations.java:109) ~[paper-1.21.1.jar:1.21.1-37-ba1b016]
	at io.papermc.paper.configuration.PaperConfigurations.initializeGlobalConfiguration(PaperConfigurations.java:[20](https://mclo.gs/6rp0D7Z#L20)3) ~[paper-1.21.1.jar:1.21.1-37-ba1b016]
	at net.minecraft.server.dedicated.DedicatedServer.initServer(DedicatedServer.java:226) ~[paper-1.21.1.jar:1.21.1-37-ba1b016]
	at net.minecraft.server.MinecraftServer.runServer(MinecraftServer.java:1[21](https://mclo.gs/6rp0D7Z#L21)5) ~[paper-1.21.1.jar:1.21.1-37-ba1b016]
	at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:330) ~[paper-1.21.1.jar:1.21.1-37-ba1b016]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[?:?]
```
Full log: https://mclo.gs/6rp0D7Z
Crash report: https://mclo.gs/uUwvRwO

### Plugin and Datapack List

None

### Actions to reproduce (if known)

1. Create a server with build 36
2. Start it with build 37

### Paper version

This server is running Paper version 1.21.1-37-master@ba1b016 (2024-08-19T10:00:18Z) (Implementing API version 1.21.1-R0.1-SNAPSHOT)
You are running the latest version
Previous version: 1.21.1-36-dae906b (MC: 1.21.1)

### Other

Deleting the config folder allows the server to start again

**Comment by lynxplay** [comment]
So sorry, my mistake, that part of the diff got lost while I rebased the PR onto latest master. Fix will be out ASAP.

**Comment by lynxplay** [comment]
Resolved by e4b38b4f8c2d6647a36865b64f63df771b5e4041, again, sorry for the inconvenience.

**Comment by JulianVennen** [comment]
That did fix it, thank you for the fast response.
Hope you have a nice day :)
